MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m

```diff
@@ -146,32 +154,85 @@ + (BOOL)isAutoPageTrackingEnabled {
 
 #pragma mark - Private methods
 
-- (BOOL)validateProperties:(NSDictionary<NSString *, NSString *> *)properties {
+- (BOOL)validateEventName:(NSString *)eventName {
+  if (!eventName || [eventName length] < minEventNameLength) {
+    MSLogError([MSAnalytics logTag],
+               @"Event name cannot be null or empty");
+    return NO;
+  }
+  if ([eventName length] > maxEventNameLength) {
+    MSLogError([MSAnalytics logTag],
+               @"Event name length cannot be longer then %d characters", minEventNameLength);
+    return NO;
+  }
+  return YES;
+}
+
+- (NSDictionary<NSString *, NSString *> *)validateProperties:(NSDictionary<NSString *, NSString *> *)properties {
+  NSMutableDictionary<NSString *, NSString *> *validProperties = [NSMutableDictionary new];
   for (id key in properties) {
     if (![key isKindOfClass:[NSString class]] || ![properties[key] isKindOfClass:[NSString class]]) {
-      return NO;
+      continue;
+    }
+
+    // Validate key
+    NSString *strKey = key;
+    if ([strKey length] < minPropertyKeyLength) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"Property key cannot be empty. Property will be skipped.");
+      continue;
+    }
+    if ([strKey length] > maxPropertyKeyLength) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"Property key length cannot be longer than %d characters. Property %@ will be skipped.",
+                   maxPropertyKeyLength,
+                   strKey);
+      continue;
+    }
+
+    // Validate value
+    NSString *value = properties[key];
+    if(value && [value length] > maxPropertyValueLength) {
+      MSLogWarning([MSAnalytics logTag],
+                   @"Property value length cannot be longer than %d characters. Property %@ will be skipped.",
```

Comment:
For all validation messages we were trying to harmonize them between Android and UWP, forgot to link the comment in the iOS PR, please use the messages from https://github.com/Microsoft/mobile-center-sdk-android/pull/391 (keep warnings though, you are right about log level).